src/CADL.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { getOperationLink } from "@azure-tools/cadl-azure-core";
+import {
+    createDpgContext,
+    isApiVersion,
+    shouldGenerateConvenient,
+    shouldGenerateProtocol
+} from "@azure-tools/cadl-dpg";
+import {
+    EmitContext,
+    getDeprecated,
+    getDoc,
+    getSummary,
+    isErrorModel,
+    Model,
+    ModelProperty,
+    Namespace,
+    Operation,
+    Program
+} from "@cadl-lang/compiler";
+import { getResourceOperation, ResourceOperation } from "@cadl-lang/rest";
+import {
+    HttpOperation,
+    HttpOperationParameter,
+    HttpOperationResponse
+} from "@cadl-lang/rest/http";
+import { NetEmitterOptions } from "../options.js";
+import { BodyMediaType, typeToBodyMediaType } from "../type/BodyMediaType.js";
+import { collectionFormatToDelimMap } from "../type/CollectionFormat.js";
+import { HttpResponseHeader } from "../type/HttpResponseHeader.js";
+import { InputConstant } from "../type/InputConstant.js";
+import { InputOperation } from "../type/InputOperation.js";
+import { InputOperationParameterKind } from "../type/InputOperationParameterKind.js";
+import { InputParameter } from "../type/InputParameter.js";
+import {
+    InputEnumType,
+    InputListType,
+    InputModelType,
+    InputType,
+    isInputLiteralType,
+    isInputUnionType
+} from "../type/InputType.js";
+import { OperationFinalStateVia } from "../type/OperationFinalStateVia.js";
+import { OperationLongRunning } from "../type/OperationLongRunning.js";
+import { OperationPaging } from "../type/OperationPaging.js";
+import { OperationResponse } from "../type/OperationResponse.js";
+import {
+    RequestLocation,
+    requestLocationMap
+} from "../type/RequestLocation.js";
+import {
+    parseHttpRequestMethod,
+    RequestMethod
+} from "../type/RequestMethod.js";
+import { getExternalDocs, getOperationId, hasDecorator } from "./decorators.js";
+import {
+    getDefaultValue,
+    getEffectiveSchemaType,
+    getInputType
+} from "./model.js";
+import { capitalize } from "./utils.js";
+
+export function loadOperation(
+    context: EmitContext<NetEmitterOptions>,
+    operation: HttpOperation,
+    uri: string,
+    urlParameters: InputParameter[] | undefined = undefined,
+    serviceNamespaceType: Namespace,
+    models: Map<string, InputModelType>,
+    enums: Map<string, InputEnumType>
+): InputOperation {
+    const program = context.program;
+    const dpgContext = createDpgContext(context);
+    const {
+        path: fullPath,
+        operation: op,
+        verb,
+        parameters: cadlParameters
+    } = operation;
+    console.log(`load operation: ${op.name}, path:${fullPath} `);
```

Comment:
As you are working on the refactor, one general comment is to add more logging information during the emitter run.
We could by default turn off the log, but if some issues happen, we could enable more logging for trouble shooting.

Reply:
add logger to do this. and an emitter option `logLevel` to set the level.